### PR TITLE
SSCS-9432 - Postponement request

### PR DESCRIPTION
**Pull request by ryandac**
SSCS-9432 - Postponement request

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestMidEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+        final PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        final String requestDetails = sscsCaseData.getPostponementRequest().getPostponementRequestDetails();
+
+        if (isEmpty(requestDetails)) {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestMidEventHandler.java:48]: is it possible to have a "" i.e blank detail?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_START)
+                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+                && callback.getCaseDetails() != null
+                && State.HEARING.equals(callback.getCaseDetails().getState());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java:26]: Do you need to check the state here? Isn't that just controlled in the CCD definition?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToSubmitHandler.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_SUBMIT)
+                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+                && callback.getCaseDetails() != null
+                && State.HEARING.equals(callback.getCaseDetails().getState());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToSubmitHandler.java:29]: Optional: Same comment here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestMidEventHandler.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+                && callback.getCaseDetails() != null
+                && State.HEARING.equals(callback.getCaseDetails().getState());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestMidEventHandler.java:38]: Optional: Same comment here

**Review by jackmaloney — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java (hunk 2 above)
Comment (ryandac) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java:26]: Yes, you're right. It is controlled by the ccd definition. We don't need the extra check here

**Pull request merged**